### ISSUE TITLE
ValidationAttribute not working as expected on Blazor server form

### ISSUE THREAD

**Issue by sipi41** [dotnet/aspnetcore#38341]
I think I found a bug while trying to implement custom validation to my daughter's dog grooming website form using Blazor server, beforehand Thank you for any help you can provide.  

THE PROBLEM is that the form, suddenly ignores my custom validation rule, please check my code and result below:

I have a class with a field like this:

`public class Pet
    {       
        [Required(AllowEmptyStrings = false, ErrorMessage = "Please enter the Pet's Name"), MaxLength(100)]
        [Test2]
        public string PetName { get; set; }
}`

(sorry for the not indented code), basically, its a class called Pet with a property called PetName which uses the Required attribute, and another attribute I created called "Test2"

Now, I created my custom validation class that ALWAYS returns an error, no matter what the content is (just to test) as follow:

`public class Test2 : ValidationAttribute
    {        
        protected override ValidationResult? IsValid(object? value, ValidationContext validationContext)
        {
            return new ValidationResult("Error here always!");
        }
    }`

Now on my form I have this as follow:
`<div class="mb-3 col-md-6">
                <label for="PetName" class="form-label">Pet's Name:</label>
                <InputText @bind-Value="Pet.PetName" class="form-control" id="PetName"></InputText>
                <ValidationMessage For="() => Pet.PetName"></ValidationMessage>
            </div> `

When I run all this, I get the form as follow:
![image](https://user-images.githubusercontent.com/688562/141514134-e91ac1ca-2534-4d8c-baba-893140a72c02.png)
If I click the submit button once, the first attribute (in this case REQUIRED) enter into action as follow:
![image](https://user-images.githubusercontent.com/688562/141514304-13a5b4a0-2939-4bd2-acc4-fb24cc5e0289.png)
Now, after I enter a value, the Custom Validation (that always return an error, as expected) is fired on change, which is totally fine, as follow:
![image](https://user-images.githubusercontent.com/688562/141514658-1f4b0859-88bd-44bc-8fd1-37837245537f.png)
THE PROBLEM comes when I hit the submit button again, (as you may expect, I have more fields in my form), the custom validation disappears, in other words, the last time it was saying something was wrong (which is totally fine as this is the way we did design the custom validation, as an example), but after pressing the submit button it is saying it complies with all the validation rules/attributes we have assigned but no, please check the following image:
![image](https://user-images.githubusercontent.com/688562/141515244-50d234af-0636-416b-af15-b7473d3573d3.png)
In other words, when the field changed it did detect that something was not ok and displayed the message, awesome! after the submit button is pressed now it changed and say everything is fine, as if when pressing the submit button it ignores any custom validation attributes... am I missing something? maybe forgetting about doing something? Thank you again.

**Comment by pranavkm** [comment]
> public class Test2 : ValidationAttribute { protected override ValidationResult? IsValid(object? value, ValidationContext validationContext) { return new ValidationResult("Error here always!"); } }

See https://docs.microsoft.com/aspnet/core/blazor/forms-validation#custom-validation-attributes

> To ensure that a validation result is correctly associated with a field when using a custom validation attribute, pass the validation context's MemberName when creating the ValidationResult.

**Comment by sipi41** [comment]
Thank you @pranavkm, it should ALWAYS include the MemberName when returning the ValidationResult. The final code should be as follow:

`public class Test2 : ValidationAttribute
    {        
        protected override ValidationResult? IsValid(object? value, ValidationContext validationContext)
        {

            return new ValidationResult("Error msg here!", memberNames: new List<string>{ validationContext.MemberName });
        }
    }`
